spec/index.bs

```diff
@@ -1034,44 +1338,28 @@ partial interface HTMLObjectElement : HTMLElement {
 
 Issue: Add base.href enforcement.
 
-### Enforcement for script text contents ### {#enforcement-in-script-text}
+### Enforcement for scripts ### {#enforcement-in-script-text}
 
 This document modifies how {{HTMLScriptElement}} [=child text content=] can be set to allow applications to control dynamically created scripts. It does so by
 adding the {{HTMLElement/innerText}} and {{Node/textContent}} attributes directly on {{HTMLScriptElement}}. The behavior of the attributes remains the same
 as in their original counterparts, apart from additional behavior triggered by the {{TrustedTypes}} extended attribute presence.
 
+Note: Using these IDL attributes is the recommended way of dynamically setting URL or a text of a script. Manipulating attribute nodes or text nodes directly will call a default policy on the final value when the script is prepared.
+
+Issue: Figure out what to do with script.setAttribute('src')
+
 <pre class="idl">
 partial interface mixin HTMLScriptElement : HTMLElement {
  [CEReactions, TrustedTypes=TrustedScript] attribute [TreatNullAs=EmptyString] ScriptString innerText;
  [CEReactions, TrustedTypes=TrustedScript] attribute ScriptString? textContent;
+ [CEReactions, TrustedTypes=TrustedScriptURL] attribute ScriptURLString src;
+ [CEReactions, TrustedTypes=TrustedScript] attribute ScriptString text;
 };
 </pre>
 
-Additionally, this document defines [$insert text node validation steps$] for {{HTMLScriptElement}} to ascertain that text nodes inserted under {{HTMLScriptElement}} elements will satisfy Trusted Types restrictions:
-
-1.  Let |text| be |node|'s [=child text content=].
-1.  Let |convertedText| be the result of running [$Get Trusted Type compliant string$], passing |parent|'s [=relevant global object=] as |global|, {{TrustedScript}} as |expectedType|, |text| as |input| and `"script.text"` as |sink|.
-    If the algorithm threw an error, rethrow the error.
-
-    Note: If Trusted Types are not enforced, this algorithm will return early and return |input| unmodified.
-1.  Set |node|'s data to |convertedText|.
+On setting, the {{HTMLElement/innerText}}, {{Node/textContent}} and {{HTMLScriptElement/text}} IDL attributes perform the regular steps, and then set [=content object=]'s {{script/[[ScriptText]]}} internal slot value with script's [=child text content=].
```

Comment:
I think you want to use the given value here and not the script's child text content. I worry somewhat that a mutation event could have modified the actual value. (Did not try to verify.)

Reply:
Sounds good.